server/endpoints/AdminActions.ts

```diff
@@ -184,13 +222,61 @@ export const resetProfessors: Endpoint<AdminProfessorsRequest> = {
 };
 
 export const reportReview: Endpoint<ReviewRequest> = {
-  guard: [body("id").notEmpty().isAscii()],
+  guard: [body("id").notEmpty().isAscii(), body("netId").notEmpty().isAscii()],
   callback: async (ctx: Context, request: ReviewRequest) => {
     try {
-      await Reviews.updateOne({ _id: request.id }, { $set: { visible: 0, reported: 1 } });
-      const course = (await Reviews.findOne({ _id: request.id })).class;
-      const res = await updateCourseMetrics(course);
-      return { resCode: res.resCode };
+      const student = await Students.findOne({ netId: request.netId });
+      if (student !== null) {
+        const { lastReported1, numReported } = student;
+        console.log(student);
+        console.log(lastReported1);
+        const lastReportedTime = lastReported1.getTime();
+        const oneDay = lastReportedTime + 1 * 24 * 60 * 60 * 1000;
+
+        if (numReported >= 3 && lastReportedTime > oneDay) {
```

Comment:
I dont think this works. I'm thinking. Its invalid if curtime <= lastReported + 24hrs